File: src/main/java/com/indiepost/dto/post/PostDto.java
package com.indiepost.dto.post;

import com.indiepost.dto.ContributorDto;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by jake on 17. 1. 22.
 */
public class PostDto extends PostSummaryDto implements Serializable {

    private String content;

    private List<String> tags = new ArrayList<>();

    private List<ContributorDto> contributors = new ArrayList<>();

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public List<String> getTags() {
        return tags;
    }

    public void setTags(List<String> tags) {
        this.tags = tags;
    }

    public List<ContributorDto> getContributors() {
        return contributors;
    }

    public void setContributors(List<ContributorDto> contributors) {
        this.contributors = contributors;
    }
}

File: src/main/kotlin/com/indiepost/dto/post/PostDto.kt
package com.indiepost.dto.post

import com.indiepost.dto.ContributorDto
import com.indiepost.dto.ProfileDto
import java.io.Serializable
import java.time.Instant
import java.util.*

/**
 * Created by jake on 17. 1. 22.
 */
data class PostDto(
        var content: String? = null,

        var tags: List<String> = ArrayList(),

        var contributors: List<ContributorDto> = ArrayList(),

        var profiles: List<ProfileDto> = ArrayList(),

        val lastRequested: Instant = Instant.now()
) : PostSummaryDto(), Serializable

